v12x/src/main/java/com/foursoft/harness/vec/v12x/navigations/SpecificationNavs.java

```diff
@@ -73,4 +64,69 @@ private SpecificationNavs() {
         };
     }
 
+    public static Function<HasSpecifications<VecSpecification>, Stream<VecOccurrenceOrUsage>> allOccurrenceOrUsages() {
+        return hasSpecifications -> Stream.concat(
+                components().apply(hasSpecifications).stream(),
+                hasSpecifications.getSpecificationsWithType(VecPartUsageSpecification.class)
+                        .stream()
+                        .map(VecPartUsageSpecification::getPartUsages)
+                        .flatMap(Collection::stream));
+    }
+
+    /**
+     * Gets the {@link VecCompositionSpecification}s and gets
+     * their {@link VecCompositionSpecification#getComponents() components}.
+     *
+     * @return A possibly-empty list of Components.
+     */
+    public static Function<HasSpecifications<VecSpecification>, List<VecPartOccurrence>> components() {
+        return hasSpecifications -> hasSpecifications
```

Comment:
hasSpecifications -> specification

Reply:
or pass the interface HasSpecifications

Reply:
I don't see how I could pass the Interface to that method, then it wouldn't be a `Function` anymore. I believe the rename is just fine.